🧑‍🌾 `ros2topic.ros2topic.test.test_bw_delay_hz.test_bw_delay_hz` failing in Kilted Zenoh

### Generated by Generative AI

_No response_

### Operating System:

Ubuntu Noble

### ROS version or commit hash:

Kilted

### RMW implementation (if applicable):

Zenoh

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

_No response_

### 'ros2 doctor --report' output

_No response_

### Steps to reproduce issue

1. Run a build in Kci Zenoh
2. See test regression fail

### Expected behavior

Test should not fail

### Actual behavior

Test is failing

### Additional information

Reference build:
- [Kci__nightly-zenoh_ubuntu_noble_amd64#35](http://build.ros2.org/job/Kci__nightly-zenoh_ubuntu_noble_amd64/35)

Test regression:
- [ros2topic.ros2topic.test.test_bw_delay_hz.test_bw_delay_hz](https://build.ros2.org/job/Kci__nightly-zenoh_ubuntu_noble_amd64/35/testReport/ros2topic.ros2topic.test/test_bw_delay_hz/test_bw_delay_hz/)

Log output:
```
FAIL: test_bw_delay_hz.TestROS2TopicBwDelayHz.test_bw_basic (topic='/clitest/topic/bw_compatible_qos', provide_qos=True, compatible_qos=True)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/ws/src/ros2/ros2cli/ros2topic/test/test_bw_delay_hz.py", line 170, in helper_verb_basic
    assert re.search(success_regex, command.output, flags=re.MULTILINE), (
AssertionError: bw CLI did not print expected message
assert None
 +  where None = <function search at 0x79acf8335c60>('^[0-9]+ B/s from [0-9]+ messages$', 'WARNING: topic [/clitest/topic/bw_compatible_qos] does not appear to be published yet\n', flags=re.MULTILINE)
 +    where <function search at 0x79acf8335c60> = re.search
 +    and   'WARNING: topic [/clitest/topic/bw_compatible_qos] does not appear to be published yet\n' = <launch_testing.tools.process.ProcessProxy object at 0x79ace8107110>.output
 +    and   re.MULTILINE = re.MULTILINE
```

Last fail on Rolling was on 2025-07-08 ([reference](https://build.ros2.org/job/Rci__nightly-zenoh_ubuntu_noble_amd64/112/#showFailuresLink)). Maybe we need to apply RMW Isolation? https://github.com/ros2/ros2cli/pull/1062

--- a/ros2topic/test/test_bw_delay_hz.py
+++ b/ros2topic/test/test_bw_delay_hz.py
@@ -145,6 +145,15 @@ class TestROS2TopicBwDelayHz(unittest.TestCase):
 
                 publish_timer = self.node.create_timer(0.5, publish_message)
 
+                # Wait for the publisher to be discovered
+                publisher_count = 0
+                timeout_count = 0
+                while publisher_count == 0 and timeout_count < 10:
+                    self.executor.spin_once(timeout_sec=0.1)
+                    publisher_count = self.node.count_publishers(topic)
+                    timeout_count += 1
+                assert publisher_count > 0, 'Publisher was not discovered'
+
                 try:
                     command_action = ExecuteProcess(
                         cmd=(['ros2', 'topic', verb] +

